### PR TITLE
feat(perf): sample dbench files for benchmark tests

### PR DESCRIPTION
**Pull request by kmova**
This PR includes some sample YAMLs using the
logdna/dbench fio scripts for running bench marking
tests using cstor and jiva engines. Also includes
YAMLs for running with host-path volumes.

This is an initial commit of the files that can be
further refined as new tunables are added in the
coming versions

Signed-off-by: kmova <kiran.mova@openebs.io>

### CODE REVIEW

File: k8s/demo/dbench/README.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  multiple volumes. 
+- The Replica instance needs to be given access to a raw block device as 
+  opposed to an `ext4` mounted path.
+
+Please refer to the https://docs.openebs.io for instantiating the cStor Pool
+(aka Replica instances). 
+
+Note: There is a cstor-sparse-pool that is launched by default as part of the
+openebs installation, which will make use of sparse files from the OS disk. 
+Run performance benchmark tests on cstor-sparse pool can result in OS
```
Comment (AmitKumarDas) [inline k8s/demo/dbench/README.md]: Running vs Run?

Can we highlight this entire sentence?

File: k8s/demo/dbench/README.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+There is also going to be support for specifying `TargetNodeSelector` for 
+cStor Targets in the coming releases. However, if you are running 0.8.0, 
+to pin the target to a node, its deployment needs to be edited
+to specify the `nodeSelector`.
+
+cStor target also allows for tuning the number of IO worker threads (called
+`LU Workers`) and queue depth. For sequential workloads, having a single 
+LU worker performs better. These tunables will be exposed via the storage
+policies similar to the replica count in 0.9. These values are translated
```
Comment (AmitKumarDas) [inline k8s/demo/dbench/README.md]: Can we tie this up with a github issue?

File: k8s/demo/dbench/README.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+into a configuration file (istgt.conf) that resides within the target pod. 
+To modify them, 
+ `kubectl exec -it -n openebs <cstor-target-pod> -c cstor-istgt /bin/bash`
```
Comment (AmitKumarDas) [inline k8s/demo/dbench/README.md]: Can we put these lines within ```bash

Comment (AmitKumarDas) [reply k8s/demo/dbench/README.md]: We can use $> & bash comments as well if we write it within ```bash

File: k8s/demo/dbench/README.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
+This folder contains sample YAMLs for running bench marking tests
```
Comment (AmitKumarDas) [inline k8s/demo/dbench/README.md]: Can we have folder structure as:
 `k8s/demo/perf/dbench`
or 
`k8s/perf/dbench`
or 
`perf/dbench`

File: k8s/demo/dbench/dbench-hostpath.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  name: dbench-hp
+spec:
+  template:
+    spec:
+      containers:
+      - name: dbench-hp
+        image: logdna/dbench:latest
+        imagePullPolicy: Always
+        env:
+          - name: DBENCH_MOUNTPOINT
```
Comment (AmitKumarDas) [inline k8s/demo/dbench/dbench-hostpath.yaml]: Is it possible to explain what each of these env variable is meant to do?
In other words yaml comments !!

File: k8s/demo/dbench/dbench-hostpath.yaml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+          # - name: FIO_DIRECT
+          #   value: "0"
+        volumeMounts:
+        - name: dbench-pv
+          mountPath: /data
+      restartPolicy: Never
+      volumes:
+      - name: dbench-pv
+        hostPath:
+          path: /var/openebs/dbench-data
```
Comment (AmitKumarDas) [inline k8s/demo/dbench/dbench-hostpath.yaml]: Can we put some comments here?
Will this be `/mnt/disks/ssd0/dbench-data`

File: k8s/demo/dbench/dbench-jdefault.yaml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 5Gi
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: dbench-jd
```
Comment (AmitKumarDas) [inline k8s/demo/dbench/dbench-jdefault.yaml]: what is `-jd`?

File: k8s/demo/dbench/dbench-jdefault-r1.yaml [hunk 8]
```diff
@@ -0,0 +1,55 @@
+kind: PersistentVolumeClaim
```
Comment (AmitKumarDas) [inline k8s/demo/dbench/dbench-jdefault-r1.yaml]: I am kind of lost with these yamls?
Can we have folder structure that help reader understand better?
e.g.
`perf/0.8.0/dbench/hostpath/`
`perf/0.8.0/dbench/jiva-pv/`
`perf/0.8.0/dbench/cstor-gpd/`
`perf/0.8.0/dbench/cstor-local/`

Comment (AmitKumarDas) [reply k8s/demo/dbench/dbench-jdefault-r1.yaml]: How are we going to showcase performance variation across openebs versions?

**Comment by kmova** [comment]
Thanks @AmitKumarDas. I will incorporate these comments in another PR shortly.

**Pull request merged**